### PR TITLE
[IMP] Increasing Coverage & Migrating to New api AFR

### PR DESCRIPTION
**Pull request by hbto**
[IMP] Increasing Coverage & Migrating to New api AFR

### CODE REVIEW

File: account_financial_report/model/account_financial_report.py [hunk 1]
```diff
@@ -43,72 +43,48 @@ class AccountFinancialReport(models.Model):
...
+        '''When report type changes boolean field analytic ledger changes
+        if applies'''
+        for brw in self:
+            if brw.inf_type != 'BS':
+                brw.update({'analytic_ledger': False})
 
-        if inf_type != 'BS':
-            res['value'].update({'analytic_ledger': False})
+    @api.onchange('columns', 'fiscalyear_id')
+    def onchange_columns(self):
```
Comment (nhomar) [inline account_financial_report/model/account_financial_report.py]: Can you add the docstring as the otheres, they are pretty clear.

File: account_financial_report/model/account_financial_report.py [hunk 2]
```diff
@@ -43,72 +43,48 @@ class AccountFinancialReport(models.Model):
...
-            period_ids = p_obj.search(cr, uid,
-                                      [('fiscalyear_id', '=', fiscalyear_id),
-                                       ('special', '=', False)],
-                                      context=context)
-            res['value'].update({'period_ids': period_ids})
-        else:
-            res['value'].update({'period_ids': []})
-        return res
+    @api.onchange('analytic_ledger')
+    def onchange_analytic_ledger(self):
```
Comment (nhomar) [inline account_financial_report/model/account_financial_report.py]: Same here

File: account_financial_report/model/account_financial_report.py [hunk 3]
```diff
@@ -43,72 +43,48 @@ class AccountFinancialReport(models.Model):
...
-            cr, uid, company_id, context=context).currency_id.id
-        fy_id = self.pool.get('account.fiscalyear').find(
-            cr, uid, context=context)
-        res['value'].update({'fiscalyear_id': fy_id})
-        res['value'].update({'currency_id': cur_id})
-        res['value'].update({'account_ids': []})
-        res['value'].update({'period_ids': []})
-        return res
+    @api.onchange('company_id')
+    def onchange_company_id(self):
```
Comment (nhomar) [inline account_financial_report/model/account_financial_report.py]: same here

**Comment by nhomar** [comment]
really nice job. ;-) just few comments but not deep ones.

**Comment by nhomar** [comment]
In order to increase the coverage you should assert the onchange values in your tests if not they will not be triggered.

**Comment by hbto** [comment]
@nhomar, you told me, on changes will be triggered when instantiating wizards, that is another tune you are playing on :-/

Though I was not intending  this PR to be reviewed yet

**Comment by nhomar** [comment]
Yes.

But you need assert the field(s) because if not it is not triggered. sorry I miss that part.

**Pull request merged**